### ISSUE TITLE
(Documentation) glob pattern support for stories config option

### ISSUE THREAD

**Issue by matthias-ccri** [storybookjs/storybook#10153]
Hi folks! Thanks for the awesome product.

TLDR: This issue is a request to document the fact that storybook uses glob for the `stories` config.

It seems like the `stories` option in `.storybook/main.js` is being parsed by [glob](https://www.npmjs.com/package/glob), but I don't think this is documented. 

In the [React Guide](https://storybook.js.org/docs/guides/guide-react/#step-3-create-the-main-file), it hints at some pattern matching support but doesn't exactly say what's supported:

Under "Step 3: Create the main file", it uses a pattern: 
```js
module.exports = {
stories: ['../src/**/*.stories.[tj]s'],
};
```

P.S. My use case is that I wanted to match all of these file extensions: js, jsx, ts, tsx. I was able to do it with this pattern: `../src/**/*.stories.{js,ts,jsx,tsx}`

**Comment by ndelangen** [comment]
It seems this page is the right place to place this info:
https://github.com/storybookjs/storybook/blob/next/docs/src/pages/configurations/overview/index.md

I think the page is missing on current docs and also on next docs
https://storybook.js.org/docs/basics/introduction/
https://deploy-preview-93--storybook-frontpage.netlify.com/docs/basics/introduction/

We should fix that. @matthias-ccri would you be able to assist?

**Comment by matthias-ccri** [comment]
Sure thing — here's a PR: https://github.com/storybookjs/storybook/pull/10196

**Comment by ndelangen** [comment]
Thank you!

**Comment by Domiii** [comment]
This information from this issue is outdated, and at the same time, one of the top google results. If you (like me) are confused about Storybook's `glob` resolution algorithm:

1. The official documentation on this is now [here](https://storybook.js.org/docs/react/configure/overview#configure-story-loading)
2. As explained on that page, the glob syntax is that of [micromatch](https://github.com/micromatch/micromatch#extended-globbing) (and not that of [glob](https://www.npmjs.com/package/glob)).
3. Here is the storybook code that parses and resolves stories: https://github.com/storybookjs/storybook/blob/master/lib/core-common/src/utils/to-require-context.ts#L31

Hope this helps! (it took me a bit of digging to find all of that out)
